Investigate paging mismatch on some combination

And if an operation defines a basic x-ms-pageable:
```
        "x-ms-pageable": {
          "nextLinkName": null
        }
```

And a model that uses x-ms-client-name:
```
"ListIndexesResult": {
      "properties": {
        "value": {
          "x-ms-client-name": "Indexes",
          "type": "array",
          "readOnly": true,
          "items": {
            "$ref": "#/definitions/SearchIndex"
          },
          "description": "The indexes in the Search service."
        }
      },
      "required": [
        "value"
      ],
      "description": "Response from a List Indexes request. If successful, it includes the full definitions of all indexes."
    },
```

We generate the wrong code, we try to access `value` where we should access `indexes`:
```python
    async def extract_data(pipeline_response):
        deserialized = self._deserialize('ListIndexesResult', pipeline_response)
>       list_of_elem = deserialized.value
E       AttributeError: 'ListIndexesResult' object has no attribute 'value'
```

--- a/ChangeLog.md
+++ b/ChangeLog.md
@@ -1,5 +1,11 @@
 # Change Log
 
+### Unreleased
+Modelerfour version: 4.13.351
+
+**Bug Fixes**
+- Corrected generation of the item name of paging response when extracting data  #648
+
 ### 2020-05-22 - 5.0.0-preview.8
 Modelerfour version: 4.13.351
 

--- a/autorest/codegen/models/paging_operation.py
+++ b/autorest/codegen/models/paging_operation.py
@@ -77,14 +77,13 @@ class PagingOperation(Operation):
             # Default value. I still check if I find it,  so I can do a nice message.
             item_name = "value"
             try:
-                self._find_python_name(item_name, "itemName")
+                return self._find_python_name(item_name, "itemName")
             except ValueError:
                 response = self._get_response()
                 raise ValueError(
                     f"While scanning x-ms-pageable, itemName was not defined and object"
                     + f" {response.schema.name} has no array called 'value'"
                 )
-            return item_name
         return self._find_python_name(self._item_name, "itemName")
 
     @property

